### PR TITLE
test: enable integration test

### LINKED ISSUE #29: enable integration test
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

could refer to 
https://github.com/kubernetes-sigs/azurefile-csi-driver/tree/master/test/integration

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Sakuralbj**
**What type of PR is this?**
 /kind test

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #29 

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
 dist: bionic
-
+services:
+  - docker
 go:
   - "1.13.9"
 
 before_install:
   - GO111MODULE=off go get github.com/mattn/goveralls
+  - mkdir /tmp/share
```
Comment (andyzhangx) [inline .travis.yml:11]: can you make these 3 lines of scripts to be included in integration-test test scripts

File: Makefile [hunk 2]
```diff
@@ -1,143 +1,144 @@
...
+	curl https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 | bash
+
+.PHONY: e2e-teardown
+e2e-teardown:
+	helm delete smb-csi-driver --namespace kube-system
+
+.PHONY: smb
+smb:
+	CGO_ENABLED=0 GOOS=linux go build -a -ldflags ${LDFLAGS} -o _output/smbplugin ./pkg/smbplugin
+
```
Comment (andyzhangx) [inline Makefile:95]: pls only make necessary change in this file, don't change the format.

**Comment by andyzhangx** [comment]
```
E0623 09:41:52.983271     876 utils.go:116] GRPC error: rpc error: code = Internal desc = volume(volumetest) mount "//0.0.0.0/Downloads" on "/tmp/stagingtargetpath" failed with mount failed: exit status 1
Mounting command: mount
Mounting arguments: -t cifs -o username=username,password=test //0.0.0.0/Downloads /tmp/stagingtargetpath
Output: mount: only root can use "--options" option
volume(volumetest) mount "//0.0.0.0/Downloads" on "/tmp/stagingtargetpath" failed with mount failed: exit status 1
Mounting command: mount
Mounting arguments: -t cifs -o username=username,password=test //0.0.0.0/Downloads /tmp/stagingtargetpath
Output: mount: only root can use "--options" option
```

File: test/integration/run-test.sh [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+target_path='/tmp/targetpath'
+cloud='AzurePublicCloud'
+
+echo "Begin to run integration test on $cloud..."
+
+# Run CSI driver as a background service
+_output/smbplugin --endpoint "$endpoint" --nodeid CSINode -v=5 &
+trap cleanup EXIT
+
+if [[ "$cloud" == 'AzureChinaCloud' ]]; then
```
Comment (andyzhangx) [inline test/integration/run-test.sh:41]: remove this, we don't need to run on Azure in this test

File: test/integration/run-test.sh [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  sleep 25
+else
+  sleep 5
+fi
+
+# set secret for csc node stage
+export X_CSI_SECRETS=username=username,"password=test"
+
+# Begin to run CSI functions one by one
+if [[ "$cloud" != 'AzureChinaCloud' ]]; then
```
Comment (andyzhangx) [inline test/integration/run-test.sh:51]: remove this

File: test/integration/run-test.sh [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  # azure file mount/unmount on travis VM does not work against AzureChinaCloud
+  echo "stage volume test:"
+  "$CSC_BIN" node stage --endpoint "$endpoint" --cap 1,block --staging-target-path "$staging_target_path" --vol-context=source="//0.0.0.0/Downloads" "$volumeid"
```
Comment (andyzhangx) [inline test/integration/run-test.sh:54]: I think `share` makes sense
`Downloads` --> `share`

File: test/integration/run-test.sh (hunk 3 above)
Comment (Sakuralbj) [reply test/integration/run-test.sh:41]: ok!

File: .travis.yml [hunk 6]
```diff
@@ -1,16 +1,25 @@
...
+  - chmod 777 /tmp/share
+  - docker run -it --name samba  -p 445:445 -v /tmp/share:/mount -d dperson/samba -u "username;test"  -s "Downloads;/mount/;yes;no;no;all;user"
```
Comment (andyzhangx) [inline .travis.yml:13]: shall we use this official samber image: https://hub.docker.com/r/servercontainers/samba ?

Comment (Sakuralbj) [reply .travis.yml:13]: OK, I will do a test later

File: Makefile [hunk 7]
```diff
@@ -1,143 +1,143 @@
...
-.PHONY: build-push
-build-push: smb-container
-	docker tag $(IMAGE_TAG) $(IMAGE_TAG_LATEST)
-	docker push $(IMAGE_TAG_LATEST)
-
-.PHONY: clean
-clean:
-	go clean -r -x
-	-rm -rf _output
+# Copyright 2017 The Kubernetes Authors.
```
Comment (andyzhangx) [inline Makefile:1]: have you changed the format of this file?

File: Makefile [hunk 8]
```diff
@@ -1,143 +1,143 @@
...
+unit-test:
+	go test -v -race ./pkg/... ./test/utils/credentials
+
+.PHONY: sanity-test
+sanity-test: smb
+	go test -v -timeout=10m ./test/sanity
+
+.PHONY: integration-test
+integration-test: smb
+	go test -v -timeout=10m ./test/integration
```
Comment (andyzhangx) [inline Makefile:60]: try add `sudo` here:
```
=== RUN   TestIntegration
--- FAIL: TestIntegration (0.00s)
    integration_test.go:41: Integration test failed fork/exec ./test/integration/run-test.sh: permission denied
FAIL
FAIL	github.com/kubernetes-csi/csi-driver-smb/test/integration	0.005s
FAIL
Makefile:60: recipe for target 'integration-test' failed
make: *** [integration-test] Error 1
The command "make integration-test" exited with 2.
```

File: Makefile (hunk 7 above)
Comment (Sakuralbj) [reply Makefile:1]: Actually, I don’t know why this  happened. I am using the same makefile as  upstream.

**Comment by andyzhangx** [comment]
still same error:
```
E0624 11:12:11.089733   32663 mount_linux.go:150] Mount failed: exit status 1
Mounting command: mount
Mounting arguments: -t cifs -o username=username,password=test //0.0.0.0/share /tmp/stagingtargetpath
Output: mount: only root can use "--options" option
```

can you add sudo everywhere to make it work first?

File: Makefile [hunk 9]
```diff
@@ -57,6 +57,7 @@ sanity-test: smb
 
 .PHONY: integration-test
 integration-test: smb
+	chmod u+x test/integration/run-test.sh
```
Comment (andyzhangx) [inline Makefile:60]: why would you do this?

File: .travis.yml [hunk 10]
```diff
@@ -14,3 +16,4 @@ script:
   - make verify
   - make smb-windows  
   - make container
+  - make integration-test
```
Comment (andyzhangx) [inline .travis.yml:19]: I think you could add `sudo` here, refer to:
https://github.com/kubernetes-sigs/azurefile-csi-driver/blob/master/.travis.yml#L12

File: .travis.yml [hunk 11]
```diff
@@ -14,3 +16,4 @@ script:
   - make verify
   - make smb-windows  
   - make container
+  - sudo -E env "PATH=$PATH" make integration-test
```
Comment (andyzhangx) [inline .travis.yml:19]: looks like it's duplicated with Makefile, I will merge this PR first, pls refine it later.

**Pull request merged**